src/EA.Iws.RequestHandlers/NotificationMovements/BulkReceiptRecovery/ReceiptRecoveryRecoveryOnlyRule.cs

```diff
@@ -37,7 +37,10 @@ public async Task<ReceiptRecoveryContentRuleResult<ReceiptRecoveryContentRules>>
             {
                 var actualMovement = actualMovements.FirstOrDefault(p => p.Number == movement.ShipmentNumber);
 
-                if (actualMovement != null && actualMovement.Status != MovementStatus.Received)
+                if (actualMovement != null && 
```

Comment:
This is the main change of this PR. Other changes in other files are just code refactoring.